src/source.rs

```diff
@@ -0,0 +1,118 @@
+use anyhow::{anyhow, Result};
+use crossbeam_channel::{Receiver, Sender};
+use std::{path::PathBuf, thread};
+
+use compositor_common::{scene::Resolution, Frame};
+use compositor_pipeline::pipeline::PipelineSource;
+use ffmpeg_next::{
+    codec::{self, Context, Id},
+    format::{self, Pixel},
+    frame, Dictionary, Rational,
+};
+
+pub struct RtpSource {
+    #[allow(dead_code)]
+    port: u16,
+    sender: Sender<Frame>,
+}
+
+impl RtpSource {
```

Comment:
Up

Reply:
Yes, you are right I got confused by gstreamer naming convention where:
- in case of pipeline it's first element is a source and result is a sink
- in case of single element it's input pads are sink pads and output pads are source past

RtpSender/RtpReceiver is probably a lot cleaner.